ark/segmentation/marker_quantification_test.py

```diff
@@ -1,17 +1,146 @@
+    with pytest.raises(ValueError):
+        marker_quantification.compute_extra_prop_info(
+            prop_info=regionprop_info,
+            regionprops_extras=['centroid_dif', 'bad_extra'])
+
+    regionprop_extra_info = marker_quantification.compute_extra_prop_info(
+        prop_info=regionprop_info,
+        regionprops_extras=regionprops_extras
+    )
+
+    # assert we created all the columns specified
+    misc_utils.verify_same_elements(
+        provided_extras=regionprops_extras,
+        regionprop_extras_columns=regionprop_extra_info.columns.values)
+
+
+def test_get_cell_props():
+    cell_mask, channel_data = test_utils.create_test_extraction_data()
+
+    segmentation_labels = test_utils.make_labels_xarray(label_data=cell_mask,
+                                                        compartment_names=['whole_cell'])
+
+    regionprops_features = ['label', 'area', 'eccentricity', 'major_axis_length',
+                            'minor_axis_length', 'perimeter', 'centroid']
+    regionprops_names = ['label', 'area', 'eccentricity', 'major_axis_length',
+                         'minor_axis_length', 'perimeter', 'centroid-0', 'centroid-1']
+    regionprops_extras = ['centroid_dif', 'num_concavities']
+
+    # no extras list
+    cell_props = marker_quantification.get_cell_props(
+        segmentation_labels.loc['fov0', :, :, 'whole_cell'].values,
+        regionprops_features)
+
+    misc_utils.verify_same_elements(
+        all_features=regionprops_names,
+        cell_props_columns=cell_props.columns.values
+    )
+
+    # specify extras list
+    cell_props = marker_quantification.get_cell_props(
+        segmentation_labels.loc['fov0', :, :, 'whole_cell'].values,
+        regionprops_features,
+        regionprops_extras)
+
+    misc_utils.verify_same_elements(
+        all_features=regionprops_names + regionprops_extras,
+        cell_props_columns=cell_props.columns.values
+    )
+
+
+def test_assign_cell_features():
+    cell_mask, channel_data = test_utils.create_test_extraction_data()
+
+    segmentation_labels = test_utils.make_labels_xarray(
+        label_data=cell_mask,
+        compartment_names=['whole_cell']
+    )
+
+    input_images = test_utils.make_images_xarray(channel_data)
+
+    # define the names of the base features that can be computed directly from regionprops
+    regionprops_features = ['label', 'area', 'eccentricity', 'major_axis_length',
+                            'minor_axis_length', 'perimeter', 'centroid', 'coords', 'label']
+
+    # define the names of the extras
+    regionprops_extras = ['centroid_dif', 'num_concavities']
+
+    # define the names of everything
+    regionprops_names = ['label', 'area', 'eccentricity', 'major_axis_length',
+                         'minor_axis_length', 'perimeter', 'label',
+                         'centroid-0', 'centroid-1', 'centroid_dif', 'num_concavities']
+
+    # get all the cell ids, for testing we'll only use 1 cell id
```

Comment:
Can you test two cells instead of 1, and check that the output is right for each of them?